axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -719,6 +719,7 @@ public void onNext(QueryProviderInbound inboundRequest) {
                 @Override
                 public void onError(Throwable ex) {
                     logger.warn("Query Inbound Stream closed with error", ex);
+                    outboundStreamObserver = null;
```

Comment:
Sounds fair. Is there a doable test class to be made which verifies that this works?

Reply:
I could only reproduce the issue when I was running on Docker, so probably timing related.

Reply:
Shouldn't we make sure that the outbound stream is also properly "completed" before setting it to null? I'm afraid we'll be leaking connections if we don't close both sides.

Reply:
Will create a new issue for this, as this needs to be done in more places.